experimental/codepointtrie/src/codepointtrie.rs

```diff
@@ -0,0 +1,311 @@
+                    return self.trie_error_val_index();
+                };
+        }
+        // Returns data_pos == data_block (offset) +
+        //     portion of code_point bit field for last (4th) lookup
+        data_block + (code_point & SMALL_DATA_MASK)
+    }
+
+    /// Returns the position in the `data` array for the given code point,
+    /// where this code point is at or above the fast limit associated for the
+    /// trie type, `T`.
+    fn trie_above_fastmax_index(&self, code_point: u32) -> u32 {
+        if code_point >= self.header.high_start {
+            self.header.data_length - HIGH_VALUE_NEG_DATA_OFFSET
+        } else {
+            self.trie_internal_small_index(code_point) // helper fn
+        }
+    }
+
+    /// Returns the position in the `data` array for the given code point,
+    /// where this code point is below the fast limit associated for the
+    /// trie type, `T`.
+    fn trie_below_fastmax_index(&self, code_point: u32) -> u32 {
+        let index_array_pos: u32 = code_point >> FAST_TYPE_SHIFT;
+        let index_array_val: u16 =
+            if let Some(index_array_val) = self.index.get(index_array_pos as usize) {
+                index_array_val
+            } else {
+                return self.trie_error_val_index();
+            };
+        let fast_index_val: u32 = index_array_val as u32 + (code_point & FAST_TYPE_DATA_MASK);
+        fast_index_val
+    }
+
+    /// Returns the value that is associated with `code_point` in this [`CodePointTrie`].
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu_codepointtrie::planes;
+    /// let trie = planes::get_planes_trie();
+    /// assert_eq!(0, trie.get(0x41));  // 'A' as u32
+    /// assert_eq!(0, trie.get(0x13E0));  // 'Ꮰ' as u32
+    /// assert_eq!(1, trie.get(0x10044));  // '𐁄' as u32
+    /// ```
+    pub fn get(&self, code_point: u32) -> W {
+        let data_pos: u32 = if code_point <= T::FAST_MAX {
+            Self::trie_below_fastmax_index(self, code_point)
+        } else if code_point <= CODE_POINT_MAX {
+            Self::trie_above_fastmax_index(self, code_point)
+        } else {
+            self.trie_error_val_index()
+        };
+        // Returns the trie value (or trie's error value).
+        // If we cannot read from the data array, then return the associated constant
+        // ERROR_VALUE_RUST for the instance type for W: ValueWidth.
+        self.data
+            .get(data_pos as usize)
+            .unwrap_or(W::ERROR_VALUE_RUST)
+    }
+
+    /// Returns the value that is associated with `code_point` for this [`CodePointTrie`]
+    /// as a `u32`. This API method maintains consistency with the corresponding
+    /// originalICU APIs.
+    pub fn get_u32(&self, code_point: u32) -> u32 {
+        self.get(code_point).cast_to_widest()
+    }
+}
+
+/// Converts the serialized `u8` value for the trie type into a [`TrieTypeEnum`].
+pub fn get_code_point_trie_type_enum(trie_type_int: u8) -> Option<TrieTypeEnum> {
```

Comment:
Nit: This looks like it doesn't belong in library code.  I suggest moving it to the test code.

Reply:
Isn't this for reading the header while deserializing?

Reply:
I think we only need it for reading from the ICU4C dump file.  We shouldn't need it when reading from the ICU4X data.